test/unit/console_log_utils_test.dart

```diff
@@ -38,7 +41,96 @@ bool _assertsEnabled() {
 bool runtimeSupportsPropTypeWarnings() => _assertsEnabled();
 
 void main() {
-  group('recordConsoleLogs', () {
+  group('console log recording utilities', () {
+    group('startSpyingOnConsoleLogs', () {
+      test('stops recording logs once the returned function is called', () {
+        final logCalls = <String>[];
+        final stopSpying = startSpyingOnConsoleLogs(onLog: logCalls.add);
+
+        window.console.log('foo');
+        window.console.log('bar');
+        expect(logCalls, ['foo', 'bar']);
+
+        stopSpying();
+
+        logCalls.clear();
+        window.console.log('baz');
+        expect(logCalls, []);
+      });
+    });
+
+    group('spyOnConsoleLogs', () {
+      test('returns the value returned by the callback', () {
+        expect(spyOnConsoleLogs(() => 'return value', onLog: (_) {}), 'return value');
+      });
+
+      test('stops recording logs once the callback completes', () {
+        final logCalls = <String>[];
+        spyOnConsoleLogs(() {
+          window.console.log('foo');
+          window.console.log('bar');
+        }, onLog: logCalls.add);
+        expect(logCalls, ['foo', 'bar']);
+
+        logCalls.clear();
+        window.console.log('baz');
+        expect(logCalls, []);
+      });
+
+      test('does not swallow errors that occur in the callback', () {
+        expect(() {
+          spyOnConsoleLogs(() => throw ExceptionForTesting(), onLog: (_) {});
+        }, throwsA(isA<ExceptionForTesting>()));
+      });
+
+      test('calls onLog in the same zone as the spyOnConsoleLogs call', () {
+        final logCallZones = <Zone>[];
+        final testZone = Zone.current.fork()
+          ..run(() {
```

Comment:
![oh_my](https://user-images.githubusercontent.com/46691367/133162242-02eeff8b-3a29-41c9-bafa-46e291575b12.gif)
TIL you could do this haha